src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+                        m_totalAmountAvailable -= m_billMap[billNameNoSpaces].getAmountDue();
+
+                        // Add this bill to the funded bill list
+                        m_fundedBillsList.append(savedBill.getName());
+                    }
+
+                    // Otherwise if a bill has been switched to no longer be funded (in the event of a defund) but is presently in the funded bill list
+                    else if(!m_billMap[billNameNoSpaces].isFunded() && m_fundedBillsList.contains(savedBill.getName()))
+                    {
+                        // Add the bill's amount due back into the amount available as the bill no longer requires the funding set aside previously
+                        m_totalAmountAvailable += m_billMap[billNameNoSpaces].getAmountDue();
+
+                        // Remove the bill from the funded bill list
+                        m_fundedBillsList.removeOne(savedBill.getName());
+                    }
+                }
+            }
+        }
+    }
+
+    // Update the amount available line edit with the updated value set to two decimal places
+    m_amountAvailableEdit->setText(QString::number(m_totalAmountAvailable, 'f', 2));
+
+    // Clear the settings object in preparation for updating the config file
+    m_settings.clear();
+
+    // Create the funds information group
+    m_settings.beginGroup(m_FUNDS_INFORMATION_GROUP_LABEL);
+
+    // Write the total funds available out to the config file
+    m_settings.setValue(m_TOTAL_FUNDS_AVAILABLE_KEY, m_totalAmountAvailable);
+    m_settings.endGroup();
+    m_settings.sync();
+
+    // Create an iterator to iterate over the bill map
+    QMap<QString, Bill>::iterator billMapIterator;
+
+    // Iterate over the bill map
+    for(billMapIterator = m_billMap.begin(); billMapIterator != m_billMap.end(); ++billMapIterator)
+    {
+        // Retrieve the current bill being checked from the map
+        Bill currentBill = *billMapIterator;
+
+        // Set the group label as the bill name
+        m_settings.beginGroup(removeSpaces(currentBill.getName()));
+
+        // Write the bill amount due, due date, and status key-value pairs to the config file
+        m_settings.setValue(m_BILL_AMOUNT_DUE_KEY, currentBill.getAmountDue());
+        m_settings.setValue(m_BILL_DUE_DATE_KEY, currentBill.getDueDate().toString(m_DATE_STRING_FORMAT));
+        m_settings.setValue(m_BILL_FUNDING_STATUS_KEY, fundingStatusBooleanToString(currentBill.isFunded()));
+        m_settings.endGroup();
+    }
+
+    // Sync the settings object at the end
+    m_settings.sync();
+}
+
+void MainWindow::fundBillOnClick()
+{
+    // Iterate over the bill table widget rows
+    for(int row = 0; row < m_billTableWidget->rowCount(); row++)
+    {
+        // Iterate over the columns in each row
+        for(int col = 0; col < m_billTableWidget->columnCount(); col++)
```

Comment:
For this function and `defundBillOnClick()` we need to check every row to see if it's selected and only the column corresponding to the funding status. Is there a way to do this without the for loop iterating over each column?

Reply:
Nested for loop removed.